app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,56 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.order(:completion_date)
+  end
+
+  def show
+    @task = Task.find(params[:id].to_i)
+  end
+
+  def edit
+    @task = Task.find(params[:id].to_i)
+    unless @task
+      redirect_to books_path
+    end
+  end
+
+  def mark_complete
+    task = Task.find(params[:id].to_i)
+    redirect_to tasks_path unless task
+    status = (task.status == "incomplete" ? "complete":"incomplete")
```

Comment:
Good use of a ternary operator.  You could also have stored the status as a boolean and just changed the output based on the true/false value.